shaders/src/common_lighting.fs

```diff
@@ -71,6 +75,64 @@ float computeMicroShadowing(float NoL, float visibility) {
     return microShadow * microShadow;
 }
 
+/**
+ * IBL utilities
+ */
+
+vec2 IntersectAABB(vec3 rayOrigin, vec3 rayDir, vec3 boxMin, vec3 boxMax) {
+    vec3 tMin = (boxMin - rayOrigin) / rayDir;
+    vec3 tMax = (boxMax - rayOrigin) / rayDir;
+    vec3 t1 = min(tMin, tMax);
+    vec3 t2 = max(tMin, tMax);
+    float tNear = max3(t1);
+    float tFar = min3(t2);
+    return vec2(tNear, tFar);
+}
+
+// Assume: a <= b
+float GetSmallestPositive(float a, float b) {
```

Comment:
I would use `#define` instead so that you don't have to deal with the `precision` :-(

Reply:
I've manually inlined it in 6dc6210f556cee2f8ce36cfd404588c5ddb3fa2d, as there was a single call site left for it.